### ISSUE TITLE
Keep showing Song Year when Album year is different

### ISSUE THREAD

**Issue by shogun1984** [ampache/ampache#3895]
Dear Ampache Users:

I know this topic comes up again and again, but I would like to know if there is a workaround for my use case:

My catalog contains several artist orders in which the songs are perfectly tagged by year. I would like to create these individual titles as an album for each artist. E.g. "Elvis - Anthology". But this leads to the well-known problem of creating an “anthology” album for every year.

Can I somehow modify the code so that I still have all the songs with the correct release date under Artist and I only have one album under Albums (year doesn't matter)?

Thank you for your ideas

**Comment by mitchray** [comment]
On the artist page under Actions, click Show Artist Songs

Is that what you are after?

**Comment by shogun1984** [comment]
Yes, that's how it works as long as I leave the tracks as they are. But I would like to add them to an album or compilation. Unfortunately, the tag year is assigned to the album and not the song. So if I add an album tag to the tracks, Ampache treats each year separately so I have an album for each year.

Is there any way to add all tracks to an album and still have the year of the song displayed under "Artist - All Songs"?

Maybe a code change that doesn't take the year into account when assigning the album?

**Comment by lachlan-00** [comment]
The originalyear, originaldate and date tags are album tags. It references the year of release and the date of the specific version of the album.

There aren't many situations where a year references a song and there isn't a tag for the song year that i know of either. A compilation that doesn't tag their files correctly is 'wrong' and should be updated or use a custom tag that isn't being used for albums.

It does get asked semi regularly but unless the tag is used for songs specifically I'm not inclined to change their purpose

**Comment by jamesc2151** [comment]
Mine show up correctly like you want as long as the file tags are like this 

![Image](https://github.com/user-attachments/assets/52e14fb2-7d31-4197-a28b-eed6d514a342)

**Comment by lzbk** [comment]
> The originalyear, originaldate and date tags are album tags. It references the year of release and the date of the specific version of the album.
> 
> There aren't many situations where a year references a song and there isn't a tag for the song year that i know of either. A compilation that doesn't tag their files correctly is 'wrong' and should be updated or use a custom tag that isn't being used for albums.
> 
> It does get asked semi regularly but unless the tag is used for songs specifically I'm not inclined to change their purpose

I create mix-tapes and like to have the year of the song displayed, but in ampache they are spread all over the place…

<img width="1000" alt="Image" src="https://github.com/user-attachments/assets/d77322f8-f642-413f-943a-428932edff9d" />

But I think most anthologies & compilations would benefit from such a feature. BBC sessions for instance :
* [Bert Jansch's](https://bertjansch.bandcamp.com/album/bert-at-the-bbc) (the year goes to the title)
* [Stone the crows](https://i.discogs.com/Aq7__wCJ7A-X9OWzFoD4YBA9x-S0RErh4ueyOS3XgrE/rs:fit/g:sm/q:90/h:600/w:597/czM6Ly9kaXNjb2dz/LWRhdGFiYXNlLWlt/YWdlcy9SLTYwMTE0/NDUtMTY1MjQ0OTg4/NS02MzU5LmpwZWc.jpeg)
* Anyone taking the time to tag the Beatles anthology discs with the info contained in the [liner notes](https://imgur.com/a/anthology-1-booklet-ExcDw) would love that…

Shouldn't we consider that the value of the tags is contextual? Meaning: if all the files in a folder have the same album name (and the same album artist? (and the same year? (and track number/cd number completing one another ?))), then differing `originalyear` should be at the track level.

**Comment by lzbk** [comment]
> What your looking for is an extended tag called Common Language " Release time" The Date/Time when the original music file was released. Known by ID3v2 Specifications as: "TRDA" The 'Recording dates' frame is a intended to be used as complement to the "TYER", "TDAT" and "TIME" frames. E.g. "4th-7th June, 12th June" in combination with the "TYER" frame.
> 
> This is not commonly a default Tag used by any Tagging program such as Picard or MP3tag.. but in MP3tag can be set as a custom tag and if you use MediaMonkey like I do you can see this tag by adding it to custom tags their. Ampache is only detecting the ID3 tag for "TYER" The 'Year' frame is a numeric string with a year of the recording. This frames is always four characters long (until the year 10000). This is the default common tag used
> 
> Reference is from [ID3 Specs](https://id3.org/id3v2.3.0#ID3v2_extended_header)

Well, I had gone for the [`TORY`](https://id3.org/id3v2.3.0#line-449) tag as I thought it to be processed by ampache. Working mostly with FLAC, which does not use ID3, I went for `ORIGINALDATE` and `ORIGINALYEAR`, which ampache processes as you can see below:

<img width="400" alt="Image" src="https://github.com/user-attachments/assets/344450c6-7701-405a-808f-21f188b1388b" />

Considering the semantics of `TORY` / `ORIGINALDATE` (currently processed) I'd tend to think there is all the less reasons to split an album based on that field. But it is what ampache currently does. The following album has a single 2022 _date_  accross all files (`TDRC` in ID3 and `YEAR` in the flacs metadata), the original date is listed, cf. above, but each original year creates a new album) :
<img width="666" alt="Image" src="https://github.com/user-attachments/assets/c3c6947e-0922-4d8c-9139-ef539780313c" />

**Comment by jamesc2151** [comment]
Wow  that seems like a real pain.. 
 I think you and have have simular wants for the program to scan and display like we want but like lachlan said he would be changing alot to make those type of changes. 
 My main thing is artwork and grabbing  it from the files .   I just regrouped and cleaned all my files on my drives. (migrating from NTFS to ext4)  After each upgrade I down load, I re-catalog everything an have noted that if my MP3's contain the Album artist in it that becomes the default album cover.  So I'm pulling that item out of all files and following his Config method to  see how that works out.  
 I try not to bother him with my play items I come up with as he's a busy man I'm sure.  I try to help in bug finds when I find them as for the reason I do complete reloads each major version change .

**Comment by lachlan-00** [comment]
you can select your group columns now and remove year on the config file so this is now valid and fixed

**Comment by jamesc2151** [comment]
Thanks for the update. I'll be doing a totally new install soon as the HD
drive did a dump. will let you know how it goes.

**Comment by lzbk** [comment]
> you can select your group columns now and remove year on the config file so this is now valid and fixed

That's great! Thank you very much for all the hard work.

In which version is that? I just updated to the last 7 version (branch _release7_ → v. `7.10.2`) and I did not find how to set this up (searched `group`, `album`, `column` in the newly downloaded config file, looked up [ampache 7 for admins](https://ampache.org/docs/help/troubleshooting/ampache7-for-admins) and [ampache 8 for admins](https://ampache.org/docs/help/troubleshooting/ampache8-for-admins) ).

I might have missed it but if someone would point me in the right direction, I should be very grateful…

**Comment by jamesc2151** [comment]
Album grouping field in the config file

Drop a field to merge more loosely, e.g. remove barcode/catalog_number/version to stop different
; pressings of the same release from splitting into separate albums. You can even drop `name` or `year`
; if you really want to - nothing here stops you from choosing a value that merges albums you didn't
; mean to merge, so choose carefully.
